src/tools/cgget.c

```diff
@@ -746,14 +825,30 @@ int main(int argc, char *argv[])
 		goto err;
 	}
 
+#ifdef CGXGET
+	ret = parse_opts(argc, argv, &cg_list, &cg_list_len, &mode, &version);
+#else
 	ret = parse_opts(argc, argv, &cg_list, &cg_list_len, &mode);
+#endif
+	if (ret)
+		goto err;
+
+#ifdef CGXGET
+	ret = convert_cgroups(&cg_list, cg_list_len, version, CGROUP_DISK);
 	if (ret)
 		goto err;
+#endif
 
 	ret = get_values(cg_list, cg_list_len);
 	if (ret)
 		goto err;
 
+#ifdef CGXGET
+	ret = convert_cgroups(&cg_list, cg_list_len, CGROUP_DISK, version);
+	if (ret)
+		goto err;
+#endif
+
```

Comment:
Should the  `cgxget` and `cgxset` have their own files? though it will share most of the common code.  It makes life easier, though the maintenance overhead will increase. In the commit message, `cgxget -2 cpu.weight MyCgroup` -> `cgxget -2 -r cpu.weight MyCgroup`, it missing `-r` option. Can you also please update other commit messages with `-r` in the examples.

Reply:
Yeah, I think this is a good plan.  When I first added `cgxget`, there were only a couple small differences between it and `cgget`, so having them in the same file made sense.  But as you noticed, there are several `ifdefs` scattered throughout the code and it's getting ugly.

Reply:
And good catch on the erroneous comment :/